polkadot/runtime/src/genesismap.rs

```diff
@@ -52,32 +53,32 @@ impl GenesisConfig {
 	pub fn genesis_map(&self) -> HashMap<Vec<u8>, Vec<u8>> {
 		let wasm_runtime = include_bytes!("../wasm/genesis.wasm").to_vec();
 		vec![
-			(&b"gov:apr"[..], vec![].join(&self.approval_ratio)),
-			(&b"ses:len"[..], vec![].join(&self.session_length)),
-			(&b"ses:val:len"[..], vec![].join(&(self.validators.len() as u32))),
-			(&b"sta:wil:len"[..], vec![].join(&0u32)),
-			(&b"sta:spe"[..], vec![].join(&self.sessions_per_era)),
-			(&b"sta:vac"[..], vec![].join(&(self.validators.len() as u32))),
-			(&b"sta:era"[..], vec![].join(&0u64)),
+			(&b"gov:apr"[..], vec![].and(&self.approval_ratio)),
```

Comment:
wishlist: replace all these `vec![].and(x)` lines with something like `encode(x)`